frontend/app/src/components/StatusWidget/StatusWidget.tsx

```diff
@@ -136,237 +106,244 @@ const RUNNING_MAN_DISPLAY_DELAY_TIME_MS = 500
  * Displays various script- and connection-related info: our WebSocket
  * connection status, the run-state of our script, and other transient events.
  */
-class StatusWidget extends PureComponent<StatusWidgetProps, State> {
-  /** onSessionEvent signal connection */
-  private sessionEventConn?: SignalConnection
-
-  private curView?: ReactNode
-
-  private readonly minimizePromptTimer = new Timer()
+const StatusWidget: React.FC<StatusWidgetProps> = ({
+  connectionState,
+  sessionEventDispatcher,
+  scriptRunState,
+  rerunScript,
+  stopScript,
+  allowRunOnSave,
+  //theme,
+}): ReactElement => {
+  const [statusMinimized, setStatusMinimized] = useState(shouldMinimize())
+  const [scriptChangedOnDisk, setScriptChangedOnDisk] = useState(false)
+  const [promptMinimized, setPromptMinimized] = useState(false)
+  const [promptHovered, setPromptHovered] = useState(false)
+  const [showRunningMan, setShowRunningMan] = useState(false)
+  const minimizePromptTimer: React.MutableRefObject<Timer | null> =
+    useRef(null)
+  const showRunningManTimer: React.MutableRefObject<Timer | null> =
```

Comment:
nit: upon first glance, this struck me as odd that this variable wasn't a boolean. perhaps `runningManTimer`?

Reply:
I think `runningManTimer` would be slightly misleading because this timer is for a delay in showing the running man. I think `runningManTimer` would suggest that it is a Timer for how long running man is displayed or something similar to that. 

This isn't related to this PR, but I could change this to `delayShowRunningManTimer` -- what do you think? 

If so, I might do this in a separate PR since it isn't related to making this a functional component.

Reply:
yeah `delayShowRunningManTimer` sounds good!